services/core/PlatformDriverAgent/tests/test_bacnet.py

```diff
@@ -0,0 +1,209 @@
+        # below are optional; values are set to show configuration options; values use the default values
+        "max_apdu_length": 1024,
+        "object_id": 599,
+        "object_name": "Volttron BACnet driver",
+        "vendor_id": 5,
+        "segmentation_supported": "segmentedBoth",
+    }
+    bacnet_proxy_agent_uuid = volttron_instance.install_agent(
+        agent_dir=get_services_core("BACnetProxy"),
+        config_file=bacnet_proxy_agent_config,
+    )
+    gevent.sleep(1)
+    volttron_instance.start_agent(bacnet_proxy_agent_uuid)
+    assert volttron_instance.is_agent_running(bacnet_proxy_agent_uuid)
+
+    yield bacnet_proxy_agent_uuid
+
+    print("Teardown of bacnet_proxy_agent")
+    volttron_instance.stop_agent(bacnet_proxy_agent_uuid)
+
+
+@pytest.fixture(scope="module")
+def bacnet_test_agent(bacnet_proxy_agent, config_store, volttron_instance):
+    test_agent = volttron_instance.build_agent(identity="test-agent")
+
+    # create a mock callback to use with a subscription to the driver's publish publishes
+    test_agent.poll_callback = MagicMock(name="poll_callback")
+
+    # subscribe to device topic results
+    test_agent.vip.pubsub.subscribe(
+        peer="pubsub",
+        prefix=BACNET_DEVICE_TOPIC,
+        callback=test_agent.poll_callback,
+    ).get()
+
+    # give the test agent the capability to modify the platform_driver's config store
+    capabilities = {"edit_config_store": {"identity": PLATFORM_DRIVER}}
+    volttron_instance.add_capabilities(test_agent.core.publickey, capabilities)
+
+    # A sleep was required here to get the platform to consistently add the edit config store capability
+    gevent.sleep(1)
+
+    yield test_agent
+
+    print("In teardown method of query_agent")
+    test_agent.core.stop()
+
+
+@pytest.fixture(scope="module")
+def config_store_connection(volttron_instance):
+    capabilities = [{"edit_config_store": {"identity": PLATFORM_DRIVER}}]
+    connection = volttron_instance.build_connection(
+        peer=CONFIGURATION_STORE, capabilities=capabilities
+    )
+    gevent.sleep(1)
+
+    # Start the platform driver agent which would in turn start the bacnet driver
+    platform_uuid = volttron_instance.install_agent(
+        agent_dir=get_services_core("PlatformDriverAgent"),
+        config_file={
+            "publish_breadth_first_all": False,
+            "publish_depth_first": False,
+            "publish_breadth_first": False,
+        },
+        start=True,
+    )
+    gevent.sleep(2)  # wait for the agent to start and start the devices
+
+    yield connection
+
+    volttron_instance.stop_agent(platform_uuid)
+    volttron_instance.remove_agent(platform_uuid)
+    connection.kill()
+
+
+@pytest.fixture(scope="module")
+def config_store(config_store_connection):
+    # this fixture will setup a the BACnet driver that will communicate with a live BACnet device located at PNNL campus in Richland at the given device_address
+    device_address = "130.20.24.157"
```

Comment:
I don't think I like this being out here even though it's not publicly accesible.

Reply:
Yeah, I had some reservations about exposing this. I think I'll change to an env var and add some documentation on how to run this test locally.